python/amici/sbml_import.py

```diff
@@ -1501,7 +1499,105 @@ def _add_conservation_for_non_constant_species(
             for i, c in zip(cl, coefficients):
                 A[i_cl, i] = sp.Rational(c)
         rref, pivots = A.rref()
-        species_to_be_removed = set(pivots)
+
+        raw_cls = []
+        for i_cl, target_state_model_idx in enumerate(pivots):
+            # collect values for species engaged in the current CL
+            state_idxs = [i for i, coeff in enumerate(rref[i_cl, :])
+                          if coeff]
+            coefficients = [coeff for coeff in rref[i_cl, :] if coeff]
+            raw_cls.append((target_state_model_idx, state_idxs,
+                            coefficients),)
+        return raw_cls
+
+    def _get_conservation_laws_new(
```

Comment:
```suggestion
    def _get_conservation_laws_rref(
```

Reply:
Might be somewhat confusing, as the other algorithm is also doing that as a first step, but I don't have a better idea :sweat_smile: